core/amber/src/main/scala/edu/uci/ics/texera/web/resource/dashboard/user/quota/UserQuotaResource.scala

```diff
@@ -0,0 +1,254 @@
+package edu.uci.ics.texera.web.resource.dashboard.user.quota
+
+import edu.uci.ics.texera.web.SqlServer
+import edu.uci.ics.texera.web.auth.SessionUser
+import edu.uci.ics.texera.web.resource.dashboard.user.quota.UserQuotaResource.{
+  file,
+  workflow,
+  getUserCreatedFile,
+  getUserCreatedWorkflow,
+  getUserAccessedWorkflow,
+  getUserAccessedFiles,
+  getUserMongoDBSize,
+  deleteMongoCollection,
+  mongoStorage
+}
+import org.jooq.types.UInteger
+
+import java.util
+import javax.ws.rs._
+import javax.ws.rs.core.MediaType
+import edu.uci.ics.texera.web.model.jooq.generated.Tables._
+
+import scala.collection.convert.ImplicitConversions.`collection AsScalaIterable`
+import edu.uci.ics.texera.web.storage.MongoDatabaseManager
+import io.dropwizard.auth.Auth
+
+object UserQuotaResource {
+  final private lazy val context = SqlServer.createDSLContext()
+
+  case class file(
+      userId: UInteger,
+      fileId: UInteger,
+      fileName: String,
+      fileSize: UInteger,
+      uploadedTime: Long,
+      description: String
+  )
+
+  case class workflow(
+      userId: UInteger,
+      workflowId: UInteger,
+      workflowName: String
+  )
+
+  case class mongoStorage(
+      workflowName: String,
+      size: Double,
+      pointer: String,
+      eid: UInteger
+  )
+
+  def getCollectionName(result: String): String = {
+    var quoteCount = 0
+    var name = ""
+
+    for (chr <- result) {
+      if (chr == '\"') {
+        quoteCount += 1
+      } else if (quoteCount == 3) {
```

Comment:
the constant 3 looks weird to me. Can you add some comments about it?

Reply:
Added